### PR TITLE
[wallet-ext] - update copy / explorer link UI for accounts

### PR DESCRIPTION
**Pull request by mamos-mysten**
## Description 

Updates the accounts list UI to match the latest designs for copying to clipboard and links out to Explorer.

https://github.com/MystenLabs/sui/assets/122397493/cc61ca84-0338-480c-abdc-1bd8b529804e

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: apps/wallet/src/ui/app/components/IconButton.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/components/IconButton.tsx:1]: I think I usually use the button component for icons but probably better to have a dedicated component

Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/components/IconButton.tsx:1]: I think I usually use the button component for icons but probably better to have a dedicated component

File: apps/wallet/src/ui/app/components/accounts/AccountItem.tsx [hunk 1]
```diff
@@ -31,6 +35,10 @@ export const AccountItem = forwardRef<HTMLDivElement, AccountItemProps>(
 		const copyAddress = useCopyToClipboard(account?.address!, {
 			copySuccessMessage: 'Address copied',
 		});
+		const explorerHref = useExplorerLink({
+			type: ExplorerLinkType.address,
+			address: account?.address!,
```
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/components/accounts/AccountItem.tsx:40]: I don't think address/account is always available here, `!` is the best thing to use here. Probably we need to update the `useExplorerLink` to accept undefined/null

**Review by pchrysochoidis — APPROVED** [review]

**Comment by Nikhil-Mysten** [comment]
might be more of a question for @mystie711 but should the copy icon maybe have some more padding? the hover bg looks really tight when compared to the explorer link

**Comment by mystie711** [comment]
@mamos-mysten the copy and open-explorer icons should only appear on hover like Figma shows.
image

**Comment by mystie711** [comment]
Also, noticing @mamos-mysten that the default card has a shadow in the demo clip. The card shadow should only show up on hover over a card.

**Comment by mystie711** [comment]
> might be more of a question for @mystie711 but should the copy icon maybe have some more padding? the hover bg looks really tight when compared to the explorer link

Good eye @Nikhil-Mysten. 
Did not want to create a new icon just for this use case.
I plan to update the copy icon in near future for that size.

**Comment by mamos-mysten** [comment]
Made the following changes: 
- Updated so the buttons only appear on hover
- Removed card shadow for resting state

There are still some diffs from the Figma, but I'll address these in a future PR. The major issue we're having right now is that we are trying to re-use the AccountItem components in the three different contexts in which they are used: 
- AccountsList (account display on home page)
- AccountMultiSelect
- Manage Accounts page

Unfortunately, given the current designs, these components behave slightly differently depending on the context in which they are used. e.g. AccountItem in a multi-select _should_ have a shadow at rest, if it's selected while when used in the AccountsList, it should not. 

Will take another pass at these and see if we can come up with something flexible where we don't have to implement / maintain the AccountItem in three separate places.

**Pull request merged**